geocoding/generate_pincode.py

```diff
@@ -0,0 +1,109 @@
+import logging
+import math
+import pathlib
+
+import fire
+import googlemaps
+import pandas as pd
+from tqdm import tqdm
+
+logging.basicConfig(
+    filename="pincoding.log",
+    filemode="a",
+    format="%(asctime)s,%(msecs)d %(name)s %(levelname)s %(message)s",
+    datefmt="%H:%M:%S",
+    level=logging.INFO,
+)
+
+logging.info("Logger Setup Complete")
+
+
+def reverse_geocode(csv_sheet, api_key):
+    """Function which uses GMap Geocoding API to find the pincode using given latitude longitude.
+    Saves in the same CSV sheet passed as parameter.
+    
+    Args:
+        csv_sheet (str): Name of file to add pincodes in
+        api_key (str): API Key for using GMap API
+    """
+    gmaps = googlemaps.Client(key=api_key)
+    csv_sheet = pathlib.Path.cwd() / csv_sheet
+    df = pd.read_csv(csv_sheet)
+    lats, lngs = df["Latitude"].tolist(), df["Longitude"].tolist()
+    postal_codes, pincode_verify = [], []
+    is_assam = []
+    count = 0
+    nancount = 0
+    success = 0
+    index = 0
+    for lat, lng in tqdm(zip(lats, lngs)):
```

Comment:
This is hard for me to read right now given there are 3+ flags. What can we do simplify?

One way plausible way: 
Can we separate these out different cases into neater functions -- even nested function will do.

Reply:
Cleaned the code and improved readability.
Added docstrings.